.github/workflows/test.yml

```diff
@@ -44,6 +44,13 @@ jobs:
           cd example_pkg
           ./dev.py build
           ./dev.py test
+      - name: Patch Windows runner
+        if: matrix.os == 'windows-latest'
+        run: |
+          Remove-Item -Path C:\Strawberry -Recurse
+          Remove-Item -Path C:\ProgramData\Chocolatey\bin\gcc.exe
+          Remove-Item -Path C:\msys64 -Recurse -Force
+          Remove-Item -Path "C:\Program Files\LLVM" -Recurse
```

Comment:
How would this work when using devpy in NumPy? Would we need to add this CI step?

Reply:
I guess it depends which compiler you want. If you're happy with clang, presumably you can just remove gcc. These steps are only necessary to simulate having the Visual Studio compilers.

Reply:
Yup -- this isn't really a devpy thing anyway. It happens for random C/C++ projects that don't have anything to do with python, also.

Using something like https://github.com/ilammy/msvc-dev-cmd is a good idea IMO, at least for GitHub actions. It guarantees that `cl.exe` is set up correctly *and* is somewhere in PATH, and if it's in PATH then it will always win as far as Meson is concerned.

Reply:
This PR now correctly simulates an identical setup in CI to compare to the case where an end user on a personal laptop has MSVC installed, but never went and installed some other gcc/clang compiler.

i.e. it simulates the common case.

Reply:
:+1: